From b6db327993ba5f58638fae01bb99e560728959f5 Mon Sep 17 00:00:00 2001
From: Hannes Mann <hannesmann2000@gmail.com>
Date: Sat, 22 Jul 2023 00:06:37 +0200
Subject: [PATCH 2/4] egl: Allow disabling ANGLE_sync_control

---
 src/egl/drivers/dri2/platform_x11.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/egl/drivers/dri2/platform_x11.c b/src/egl/drivers/dri2/platform_x11.c
index 31f6d493f46..3113006e4dc 100644
--- a/src/egl/drivers/dri2/platform_x11.c
+++ b/src/egl/drivers/dri2/platform_x11.c
@@ -1556,8 +1556,8 @@ dri2_initialize_x11_swrast(_EGLDisplay *disp)
       if (dri2_dpy->fd_render_gpu == dri2_dpy->fd_display_gpu)
          disp->Extensions.KHR_image_pixmap = EGL_TRUE;
       disp->Extensions.NOK_texture_from_pixmap = EGL_TRUE;
-      disp->Extensions.CHROMIUM_sync_control = EGL_TRUE;
-      disp->Extensions.ANGLE_sync_control_rate = EGL_TRUE;
+      disp->Extensions.CHROMIUM_sync_control = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
+      disp->Extensions.ANGLE_sync_control_rate = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
       disp->Extensions.EXT_buffer_age = EGL_TRUE;
       disp->Extensions.EXT_swap_buffers_with_damage = EGL_TRUE;
 
@@ -1643,8 +1643,8 @@ dri2_initialize_x11_dri3(_EGLDisplay *disp)
    if (dri2_dpy->fd_render_gpu == dri2_dpy->fd_display_gpu)
       disp->Extensions.KHR_image_pixmap = EGL_TRUE;
    disp->Extensions.NOK_texture_from_pixmap = EGL_TRUE;
-   disp->Extensions.CHROMIUM_sync_control = EGL_TRUE;
-   disp->Extensions.ANGLE_sync_control_rate = EGL_TRUE;
+   disp->Extensions.CHROMIUM_sync_control = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
+   disp->Extensions.ANGLE_sync_control_rate = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
    disp->Extensions.EXT_buffer_age = EGL_TRUE;
    disp->Extensions.EXT_swap_buffers_with_damage = EGL_TRUE;
 
@@ -1760,8 +1760,8 @@ dri2_initialize_x11_dri2(_EGLDisplay *disp)
    disp->Extensions.NOK_swap_region = EGL_TRUE;
    disp->Extensions.NOK_texture_from_pixmap = EGL_TRUE;
    disp->Extensions.NV_post_sub_buffer = EGL_TRUE;
-   disp->Extensions.CHROMIUM_sync_control = EGL_TRUE;
-   disp->Extensions.ANGLE_sync_control_rate = EGL_TRUE;
+   disp->Extensions.CHROMIUM_sync_control = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
+   disp->Extensions.ANGLE_sync_control_rate = !getenv("mesa_no_sync_control") || !!strcmp(getenv("mesa_no_sync_control"), "true");
 
    dri2_set_WL_bind_wayland_display(disp);
 
-- 
2.42.0

