From cb358463800dbcf1a24533446b6fdff861483a71 Mon Sep 17 00:00:00 2001
From: Hannes Mann <hannesmann2000@gmail.com>
Date: Mon, 2 Oct 2023 12:37:30 +0200
Subject: [PATCH 5/5] radv-defaults: Do not clear VRAM for all vkd3d titles

---
 src/util/00-radv-defaults.conf | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/util/00-radv-defaults.conf b/src/util/00-radv-defaults.conf
index f928bf7dcc5..9f15991b15b 100644
--- a/src/util/00-radv-defaults.conf
+++ b/src/util/00-radv-defaults.conf
@@ -37,7 +37,6 @@ Application bugs worked around in this file:
     <device driver="radv">
         <!-- Engine workarounds -->
         <engine engine_name_match="vkd3d">
-            <option name="radv_zero_vram" value="true" />
             <option name="radv_disable_aniso_single_level" value="true" />
             <option name="radv_dgc" value="true" />
         </engine>
-- 
2.42.0

