From b716a490e1bc7a036b89be68e83bd24715b40662 Mon Sep 17 00:00:00 2001
From: Hannes Mann <hannesmann2000@gmail.com>
Date: Sat, 1 Jul 2023 03:21:45 +0200
Subject: [PATCH 1/1] xwayland: Always generate modes from values

---
 hw/xwayland/xwayland-cvt.c | 40 ++++++++------------------------------
 1 file changed, 8 insertions(+), 32 deletions(-)

diff --git a/hw/xwayland/xwayland-cvt.c b/hw/xwayland/xwayland-cvt.c
index 4248d3869..e8efcdb47 100644
--- a/hw/xwayland/xwayland-cvt.c
+++ b/hw/xwayland/xwayland-cvt.c
@@ -25,32 +25,18 @@
 
 #include <string.h>
 #include <randrstr.h>
-#include <libxcvt/libxcvt.h>
 
 #include "xwayland-cvt.h"
 
 static void
-xwayland_modeinfo_from_cvt(xRRModeInfo *modeinfo,
-                           int hdisplay, int vdisplay, float vrefresh,
-                           Bool reduced, Bool interlaced)
+xwayland_modeinfo_from_values(xRRModeInfo *modeinfo,
+                              int hdisplay, int vdisplay, float vrefresh)
 {
-    struct libxcvt_mode_info *libxcvt_mode_info;
-
-    libxcvt_mode_info =
-        libxcvt_gen_mode_info(hdisplay, vdisplay, vrefresh, reduced, interlaced);
-
-    modeinfo->width      = libxcvt_mode_info->hdisplay;
-    modeinfo->height     = libxcvt_mode_info->vdisplay;
-    modeinfo->dotClock   = libxcvt_mode_info->dot_clock * 1000.0;
-    modeinfo->hSyncStart = libxcvt_mode_info->hsync_start;
-    modeinfo->hSyncEnd   = libxcvt_mode_info->hsync_end;
-    modeinfo->hTotal     = libxcvt_mode_info->htotal;
-    modeinfo->vSyncStart = libxcvt_mode_info->vsync_start;
-    modeinfo->vSyncEnd   = libxcvt_mode_info->vsync_end;
-    modeinfo->vTotal     = libxcvt_mode_info->vtotal;
-    modeinfo->modeFlags  = libxcvt_mode_info->mode_flags;
-
-    free(libxcvt_mode_info);
+    modeinfo->width  = hdisplay;
+    modeinfo->height = vdisplay;
+    modeinfo->hTotal = hdisplay;
+    modeinfo->vTotal = vdisplay;
+    modeinfo->dotClock = hdisplay * vdisplay * vrefresh;
 }
 
 RRModePtr
@@ -60,17 +46,7 @@ xwayland_cvt(int hdisplay, int vdisplay, float vrefresh, Bool reduced,
     char name[128];
     xRRModeInfo modeinfo = { 0, };
 
-    xwayland_modeinfo_from_cvt(&modeinfo,
-                               hdisplay, vdisplay, vrefresh, reduced, interlaced);
-
-    /* Horizontal granularity in libxcvt is 8, so if our horizontal size is not
-     * divisible by 8, libxcvt will round it up, and we will advertise a wrong
-     * size to our XRandR clients.
-     * Force the width/height (i.e. simply increase blanking which should not
-     * hurt anything), keeping the rest of the CVT mode timings unchanged.
-     */
-    modeinfo.width = hdisplay;
-    modeinfo.height = vdisplay;
+    xwayland_modeinfo_from_values(&modeinfo, hdisplay, vdisplay, vrefresh);
 
     snprintf(name, sizeof name, "%dx%d",
              modeinfo.width, modeinfo.height);
-- 
2.41.0

