From 8501c6773e382e94c486ebf6fb0cdbb08e9e068b Mon Sep 17 00:00:00 2001
From: Hannes Mann <hannesmann2000@gmail.com>
Date: Wed, 28 Feb 2024 03:57:32 +0100
Subject: [PATCH 05/12] renderloop: lower framerate estimation to 10 FPS

---
 src/core/renderloop.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/renderloop.cpp b/src/core/renderloop.cpp
index 5e3a74c7fc..3e49f3edde 100644
--- a/src/core/renderloop.cpp
+++ b/src/core/renderloop.cpp
@@ -204,7 +204,7 @@ void RenderLoop::scheduleRepaint(Item *item)
     const bool vrr = d->presentationMode == PresentationMode::AdaptiveSync || d->presentationMode == PresentationMode::AdaptiveAsync;
     if (vrr && workspace()->activeWindow() && d->output) {
         Window *const activeWindow = workspace()->activeWindow();
-        if (activeWindow->isOnOutput(d->output) && activeWindow->surfaceItem() && item != activeWindow->surfaceItem() && activeWindow->surfaceItem()->frameTimeEstimation() <= std::chrono::nanoseconds(1'000'000'000) / 30) {
+        if (activeWindow->isOnOutput(d->output) && activeWindow->surfaceItem() && item != activeWindow->surfaceItem() && activeWindow->surfaceItem()->frameTimeEstimation() <= std::chrono::nanoseconds(1'000'000'000) / 10) {
             return;
         }
     }
-- 
2.44.0

